ssl/record/rec_layer_d1.c

```diff
@@ -514,6 +518,8 @@ int dtls1_read_bytes(SSL *s, int type, int *recvd_type, unsigned char *buf,
             if (SSL3_RECORD_get_length(rr) == 0)
                 SSL3_RECORD_set_read(rr);
         } else {
+            if (s->options & SSL_OP_CLEANSE_PLAINTEXT)
+                OPENSSL_cleanse(&(SSL3_RECORD_get_data(rr)[SSL3_RECORD_get_off(rr)]), n);
```

Comment:
Depending on the read sematics for datagram connections i think the _n_ should be replaced by _SSL3_RECORD_get_length(rr)_

Since when for datagram sockets when reading *up to _n_* bytes from a datagram with a *length less than _n_**. The remaining bytes are dropped and never delivered to the application.

Reply:
Sorry _n_ need to be replaced by:
_len - SSL3_RECORD_get_off(rr)_

However reading the code twice i think OpenSSL does not drop the data, so _n_ seems fine anyways